### PR TITLE
Add ttnn matmul tests to TG unit tests

### PR DESCRIPTION
**Pull request by cglagovichTT**
### Ticket
- #9476

### Problem description
- We need ttnn unit tests for TG

### What's changed
- Added 1D and 2D fractured matmul unit tests

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by cglagovichTT** [comment]
TG unit test pipeline https://github.com/tenstorrent/tt-metal/actions/runs/9548026940/job/26314476182

**Review by TT-BrianLiu — APPROVED** [review]

**Review by cfjchu — APPROVED** [review]

File: tests/ttnn/multichip_unit_tests/test_multidevice_TG.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        weights_pt,
+        dtype=ttnn.bfloat16,
+        layout=ttnn.TILE_LAYOUT,
+        device=device_mesh,
+        mesh_mapper=ShardTensorToMesh(device_mesh, dim=3),
+    )
+
+    gt = act_pt @ weights_pt
+
+    act = ttnn.to_device(act, device_mesh)
```
Comment (cfjchu) [inline tests/ttnn/multichip_unit_tests/test_multidevice_TG.py:53]: why do you need these two API calls? the `device=device_mesh` in the `ttnn.from_torch` should've moved it onto device-mesh?

**Review by ttmchiou — APPROVED** [review]

**Comment by mikevin920** [comment]
TG unit pipeline: https://github.com/tenstorrent/tt-metal/actions/runs/9750022277

**Comment by mikevin920** [comment]
T3K demo: https://github.com/tenstorrent/tt-metal/actions/runs/9763431587
T3K frequent: https://github.com/tenstorrent/tt-metal/actions/runs/9763441204
T3K unit: https://github.com/tenstorrent/tt-metal/actions/runs/9763449488
post-commit: https://github.com/tenstorrent/tt-metal/actions/runs/9763461503

**Review by aliuTT — APPROVED** [review]

File: conftest.py [hunk 2]
```diff
@@ -144,16 +144,25 @@ def device_mesh(request, silicon_arch_name, silicon_arch_wormhole_b0, device_par
     import tt_lib as ttl
 
     device_ids = ttnn.get_device_ids()
+
     try:
-        num_devices_requested = min(request.param, len(device_ids))
```
Comment (tt-rkim) [inline conftest.py:148]: Cool feature but getting a little hairy... maybe a doc string to explain what's going on and how you can now have grids?

File: tests/scripts/tg/run_tg_unit_tests.sh [hunk 3]
```diff
@@ -3,8 +3,12 @@
 set -eo pipefail
 
 run_tg_tests() {
-  # Write tests here
+
+  echo "LOG_METAL: running run_tg_unit_tests"
+
   TT_METAL_ENABLE_REMOTE_CHIP=1 ./build/test/tt_metal/unit_tests_fast_dispatch --gtest_filter="CommandQueueSingleCardFixture.*"
+
+  pytest tests/ttnn/multichip_unit_tests/test_multidevice_TG.py
```
[the remark this answers is not in the document]
Comment (tt-rkim) [reply tests/scripts/tg/run_tg_unit_tests.sh:11]: How long does the TG unit test take now w/ these changes @mikevin920

**Comment by mikevin920** [comment]
TG unit test: https://github.com/tenstorrent/tt-metal/actions/runs/9862439321

**Comment by mikevin920** [comment]
post commit: https://github.com/tenstorrent/tt-metal/actions/runs/9863265045
TG Frequent pipeline: https://github.com/tenstorrent/tt-metal/actions/runs/9863260418

**Review by tt-rkim — APPROVED** [review]

**Pull request merged**